### PR TITLE
Indexing retry

### LINKED ISSUE #207: Indexing fails from time to time due to DNS resolution failure for the backend
See for example https://github.com/quarkusio/search.quarkus.io/issues/131#issuecomment-2017381842

It seems to happen:

* During deployment, which I can sort of understand: if the app gets redeployed first, it's possible that we get this error while the backend is getting redeployed.
* Seemingly randomly... maybe during cluster updates, if those cause a redeployement?

In any case, this noise is annoying, so we should investigate how to get rid of them:

* Ideally, using some sort of no-downtime deployment for the backend. Not sure that's easy though, since we have a single node in our cluster, and 2-node clusters are kind of discouraged (split-brain, that sort of things).
* A bit less ideally, but still acceptable, we'd adjust the deployment process to redeploy the backend completely first, then the app. Then we'd only get errors in search requests, potentially.
* Worst case we can always implement some sort of retry, so that at least we don't get unnecessary "Critical" notifications.

### PR DESCRIPTION
**Pull request by yrodiere**
Fixes #207

Well, more like "works around #207 by improving resilience to openshift fucking up our backend".

Best reviewed commit by commit, because the first commit is just refactoring.

Main changes:

1. When indexing fails, we'll wait one minute, then we'll try again. We'll do that twice before actually failing and reporting the failure.
  Retry is disabled in dev/tests by default, as well as in prod when requesting reindexing through the management endpoint.
2. When we retry, we'll mark indexing as unstable, meaning we'll update the issue title, but won't try to open/close it, nor will we add a comment (thus we won't send an unnecessary notification).
3. We'll fail indexing fast: on the first failure, we cancel mass indexing.
4. We'll now always include the report in the issue description, even when the indexing succeeds or is unstable. The advantage of updating the issue description is that we don't send a notification (but the information is still available if someone needs it).

### CODE REVIEW

**Comment by yrodiere** [comment]
> We'll now always include the report in the issue description, even when the indexing succeeds or is unstable. The advantage of updating the issue description is that we don't send a notification (but the information is still available if someone needs it).

You can see how it looks there: https://github.com/yrodiere/search.quarkus.io/issues/2

**Comment by yrodiere** [comment]
@marko-bekhta I adjusted the first commit to reduce the amount of changes in that commit.

Unless you object, maybe we can merge to staging and see how it goes?

**Review by marko-bekhta — APPROVED** [review]
Hey 😃 I just finished going through your original changes 🙈 all looks good 👍🏻 

> Unless you object, maybe we can merge to staging and see how it goes?

👍🏻 let's give it a try

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -208,9 +212,12 @@ quarkus.openshift.env.vars.QUARKUS_PROFILE=prod
 quarkus.openshift.env.vars.INDEXING_QUEUE_COUNT=${INDEXING_QUEUE_COUNT}
 quarkus.openshift.env.vars.INDEXING_BULK_SIZE=${INDEXING_BULK_SIZE}
 # Initial indexing may take a while, especially the quarkus.io Git cloning
+# Tests indicate it can take ~240s in prod,
+# so we'll use 3 times as much as the limit, to allow for 3 attempts.
+# 48 * 15 = 720 = 3 * 240
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply src/main/resources/application.properties:217]: Right, I wrongly used `64` below, it was supposed to be `48`.

[the remark this answers is not in the document]
Comment (yrodiere) [reply src/main/resources/application.properties]: Fixed

**Review by marko-bekhta — APPROVED** [review]

**Comment by yrodiere** [comment]
This seems to have saved us at least one wrong failure report:

image

And otherwise seems to behave as one would expect.

So I guess we can merge into production :)

**Comment by marko-bekhta** [comment]
nice 😃 🎉

**Pull request merged**